### PR TITLE
Add doc page on 'Why xarray-beam'

### PR DESCRIPTION
**Pull request by shoyer**
This content expands upon what we have in the README.

Also includes a few other minor doc fixes.

Apologies for spurious diffs in the notebooks, apparently that's what happens when I edit them in VSCode 🤷‍♂️

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -93,3 +51,4 @@ Contributors:
 -   Stephan Hoyer
 -   Jason Hickey
 -   Cenk Gazen
+-   Alex Merose
```
Comment (alxmrs) [inline README.md:54]: :)

File: README.md [hunk 2]
```diff
@@ -17,65 +17,23 @@ multi-dimensional labeled arrays, such as:
...
 
 **🚨 Warning: Xarray-Beam is new and unpolished 🚨**
 
 Expect sharp edges 🔪 and performance cliffs 🧗, particularly related to the
 management of lazy data with Dask and reading/writing data with Zarr. We have
-used it to efficiently process 5 TB datasets. We _expect_ it to scale to PB size
-datasets but that's easier said than done. We welcome feedback and contributions
-from early adopters, and hope to have it ready for wider audience soon.
+used it to efficiently process ~25 TB datasets. We _expect_ it to scale to PB
+size datasets but that's easier said than done. We welcome feedback and
```
Comment (alxmrs) [inline README.md:28]: style nit (and my grammar may be wrong here): add a comma between "datasets" and "but".

File: docs/why-xarray-beam.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+[Xarray](http://xarray.pydata.org/en/stable/) datasets.
+
+The project aims to facilitate data transformations and analysis on large-scale
+multi-dimensional labeled arrays, such as:
+
+-   Ad-hoc computation on Xarray data, by dividing a `xarray.Dataset` into many
+    smaller pieces ("chunks").
+-   Adjusting array chunks, using the
+    [Rechunker algorithm](https://rechunker.readthedocs.io/en/latest/algorithm.html)
+-   Ingesting large multi-dimensional array datasets into an analysis-ready,
```
Comment (alxmrs) [inline docs/why-xarray-beam.md:16]: "large, multi-dimensional"

File: docs/why-xarray-beam.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    also [Pangeo Forge](https://github.com/pangeo-forge/pangeo-forge-recipes))
+-   Calculating statistics (e.g., "climatology") across distributed datasets
+    with arbitrary groups.
+
+## Our approach
+
+In Xarray-Beam, distributed Xarray datasets are represented by Beam PCollections
+of `(xarray_beam.Key, xarray.Dataset)` pairs and are manipulated by beam
+PTransforms, where {py:class}`~xarray_beam.Key` provides metadata specifying
+variables and chunk offsets for each "chunk" in the unordered PCollection.
```
Comment (alxmrs) [inline docs/why-xarray-beam.md:27]: Kind of a long sentence. How about we end at "pairs" and start a new one?

File: docs/why-xarray-beam.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+The chunking model is flexible, allowing datasets to be split across multiple
+variables and/or into orthogonal contiguous "chunks" along dimensions.
```
Comment (alxmrs) [inline docs/why-xarray-beam.md:29]: "orthogonal, contiguous"

**Review by alxmrs — APPROVED** [review]
Looks good, added some notes.

File: docs/why-xarray-beam.md (hunk 4 above)
Comment (shoyer) [reply docs/why-xarray-beam.md:27]: revised -- take another look?

Comment (alxmrs) [reply docs/why-xarray-beam.md]: That reads better to me.

**Review by alxmrs — APPROVED** [review]
LGTM

**Pull request merged**